doc/source/ray-core/objects/limiting-memory-usage.rst

```diff
@@ -0,0 +1,274 @@
+Limiting the memory usage
+=========================
+
+The application may consume a large amount of heap space and causes the node to run out of memory. When that happens the operating system will start killing worker or raylet processes, which breaks Ray. It may stall the metrics and if this happens on the head node, it may stall the :ref:`dashboard <ray-dashboard>` or other control processes and causes Ray to fail.
```

Comment:
```suggestion
If application tasks or actors consume a large amount of heap space, it can cause the node to run out of memory (OOM). When that happens the operating system will start killing worker or raylet processes, disrupting the application. OOM may also stall metrics and if this happens on the head node, it may stall the :ref:`dashboard <ray-dashboard>` or other control processes and cause the cluster to become unusable.
```